The Great Migration.

Chituka opened the pull request:
<!-- ЗДЕСЬ должно быть **подробное описание** того, что происходит в PR и зачем это нужно. PR не должен содержать изменений, о которых здесь ничего не сказано. -->
Добавление нового корабля Интеков: Калибан
Добавление нового субшаттла для Калибана: Тадпол
Добавление деградированной турбины с уменьшенной тягой
Добавление космо-вендора Интека для выбора между модом и ригом, а также модули для первых внутри
Добавление самих МОД сьютов. Обычный и Элитный. Разница в резистах, скорости и емкости для модулей (их иконки лежат в оригинальном файле, т.к. того требовал код)
Добавление нового фонарика для МОДов интеков (косметика)
Добавление отдельной профессии Honorable Vanguard и Rearguard

![image](https://github.com/user-attachments/assets/18368f21-dec7-4618-99ed-323968ee469d)

![image](https://github.com/user-attachments/assets/dc0e49e7-cbcd-47f0-b4bf-6210ed09b832)

![image](https://github.com/user-attachments/assets/3fee9971-11fb-4618-95f0-d8768546971d)
![image_2025-01-06_05-41-28](https://github.com/user-attachments/assets/3dc4eed9-e9ba-4315-8dd5-3486d38d6d1b)
![image](https://github.com/user-attachments/assets/51bb6897-258b-4d97-8d00-e1380bb9df30)
![image](https://github.com/user-attachments/assets/931188bc-2408-4b50-96da-324dc51e6990)



<!--
  Честно заполняем галочки. Чем больше галочек, тем быстрее проверять Pull Request, соответственно он быстрее будет принят.
  Чтобы отметить - ставим `x` (икс) внутри квадратных скобочек вот так: `- [x] ...`.
  Галочки можно доставлять позже по мере окончания работы над PR'ом.
-->

- [x] Pull Request полностью завершен, мне не нужна помощь чтобы его закончить.
- [x] Я внимательно прочитал все свои изменения и багов в них не нашел.
- [x] Я запускал сервер со своими изменениями локально и все протестировал.

Chituka committed:
The Great Migration.

Калибан, Тадпол, космо-вендор Интека, МОДы, добавление их иконок в оригинальные icons (того требовал код), деградированная турбина для Калибана, фонарики интека для МОДов, Профессии для Калибана и не только в виде Honorable Vanguard и Rearguard

Chituka committed:
Заполнение бюрократии

И я забыл армор буст убрать, упс

RalseiDreemuurr reviewed (changes requested):
Изменение/добавление спрайтов в коркод запрещено, плюсом я эти спрайты в глаза не видел. Жду файл со спрайтами с пингом меня в Дискорде. По мимо этого ещё и используются МОД-ы, которые в щитспавне (очевидно не просто так). Переделывай

arealbeba reviewed (dismissed):
![image](https://github.com/user-attachments/assets/316c4e55-0e54-4cce-af4e-b66932a41788)
![image](https://github.com/user-attachments/assets/99c1e71a-d5cb-4140-a529-6df30456bc67)
В туалете вашего корабля из-за бага СДММ в variable "id" у кнопки Door bolts задана буква "S", что ломает проверку мапдиффбота

XoMkA-BY commented:
Просто напомню, что МОДсьюты в текущей версии билда, как нашей, так и оффов, неиграбельны.
Да, они работают, но _**этот интерфейс и их сломанная(не знаю каким именно образом) механика - отвратительны, и потому никто не допустит их в игру вне щитспавна.**_

Chituka commented:
Модсьюты используются как фишка корабля. Я посмотрел в них всё кучу раз, интерфейс работает, модули работают, всё работает, но отсутствует. Можно мне нормальную причину, желательно которую я могу исправить, почему они в "щитспавне" (их просто нет в игре)? А добавленные спрайты используются лишь для модсьютов, на скрине есть, в дискорде продублирую.

Chituka committed:
Ну вроде фикс

Chituka committed:
Чиним проблему модсьютов

Chituka committed:
Твики Калибана по запросам игроков

Мы же хорошие разработчики, верно?

Chituka committed:
Обновление спрайтов для obj

Chituka committed:
Trauma team compatibility

Ксенофобия апдейт

Chituka committed:
Ещё твики Калибана

Когда это кончится...

Chituka commented:
Теперь кепори и воксы не могут носить МОДсьюты из-за их экспериментальности и новизны на фронтире, и сам Калибан получил изменения, согласно запросам игроков, а именно:
Были добавлены бластдуры в самый перед корабля

Были убраны стартовые ресурсы для химии, что ещё сильнее отодвинет её появление на Калибане (особенно учитывая отсутствия печей и майнинга как такового)

Также общие QoL изменения, по типу обновления субшаттла, добавление отдельной помпы для вывода атмоса и тп.

Chituka commented:
Ну и по поводу работоспособности МОДсьютов:
https://discord.com/channels/1100198143456465067/1325676693008158771/1325919072617107516

Chituka committed:
Вот до чего довела меня ваша хвалёная модульность

Иконки теперь находятся в вашей модульной папке. Это такой акт садомазохизма был, вы бы знали

Chituka commented:
> Изменение/добавление спрайтов в коркод запрещено, плюсом я эти спрайты в глаза не видел. Жду файл со спрайтами с пингом меня в Дискорде. По мимо этого ещё и используются МОД-ы, которые в щитспавне (очевидно не просто так). Переделывай

Я убил часов пять и сделал иконки вне кор-кода. Теперь выходит так, что кроме закомментирования я ВООБЩЕ не трогал кор-код. Радуемся, получается

RalseiDreemuurr commented on mod_celadon/inteq_vendor/code/inteq_MODsuit.dm:
```diff
@@ -0,0 +1,288 @@
+	name = "Elite InteQ"
+	desc = "This is an elite version of InteQ MOD suit, featuring better overall protection."
+	default_skin = "inteqelite"
+	armor = list("melee" = 60, "bullet" = 45, "laser" = 50, "energy" = 45, "bomb" = 90, "bio" = 100, "rad" = 50,"fire" = 100, "acid" = 90) //Пока оставляю резисты такими, ибо модули модсьютов нельзя сделать, тем самым их полный потенциал не реализуется.
```
![image](https://github.com/user-attachments/assets/8fe7b3ae-df43-464e-88ed-de2437de8c8f)
![image](https://github.com/user-attachments/assets/48b42759-d8d0-466f-8165-eb2c5e7b61f0)

Chituka commented:
![image](https://github.com/user-attachments/assets/3166a9d9-511c-4a77-a18b-ea93100bce56)
>Их резисты повышены относительно оригинальных модсьютов, из-за отсутствия возможности создавать модули в протолатах, мехфабах и т.п, но они обладают замедлением, которое никак не устранить. Так, обычный МОД будет замедлять до 70 процентов скорости, а элитный - до 80

Chituka commented:
![image](https://github.com/user-attachments/assets/504ac91f-48d3-4613-a722-fc749512b4f9)
А вот, ертшный, как-то резко нормально по статам, да?
![image](https://github.com/user-attachments/assets/386f4103-8a1b-452a-b971-461cc49907e4)
А ещё кое-кто очень избирательно код пастит, вот интересно! :)

Дебаг мод, насколько я помню, должен был быть на замену стандартному моду или что-то такое

Chituka committed:
Как у вас этого не было?

Нет, серьезно, почему вы просто забили на сокрытие хвостов и рогов Этереалов?

Chituka committed:
Update inteq_caliban.dmm

Косметика

Chituka committed:
Merge branch 'CeladonSS13:beta-dev' into MOD_CARIB

Chituka committed:
Я забыл про маски интека

Спасибо Quinal'у за то, что напомнил

Chituka committed:
Merge branch 'MOD_CARIB' of https://github.com/Chituka/Shiptest into MOD_CARIB

Chituka committed:
Merge branch 'CeladonSS13:beta-dev' into MOD_CARIB

Quinalion committed:
Дюфель для Интека, изменения оутфита

Quinalion commented:
![image](https://github.com/user-attachments/assets/5821cf88-88c7-439f-8b3e-c614bd64958e)

Quinalion committed:
В экстренной коробке Интек теперь их фирменная маска

Quinalion committed:
Популяризация МОДсьютов, добавил рецепты, ветки в РнД

Chituka commented:
Спасибо Куиналиону, теперь МОДы и их модули есть в ветках рнд. А ещё он интековский дюффель добавил! Вот такие дела

Chituka commented:
Резолвну потом

Chituka committed:
Merge branch 'beta-dev' into MOD_CARIB

Signed-off-by: Chituka <102897715+Chituka@users.noreply.github.com>

Chituka committed:
QoL спрайтов элитного рига

А ещё флаг на сокрытие униформы убрал, чтоб не втыкал

Chituka committed:
Merge branch 'CeladonSS13:beta-dev' into MOD_CARIB

Quinalion committed:
Морды и фурри ноги

Chituka committed:
Merge branch 'beta-dev' into MOD_CARIB

Signed-off-by: Chituka <102897715+Chituka@users.noreply.github.com>

RalseiDreemuurr commented on code/modules/mod/mod_control.dm:
```diff
@@ -624,7 +624,7 @@
 		part.remove_atom_colour(WASHABLE_COLOUR_PRIORITY)
 		part.add_atom_colour(new_color, FIXED_COLOUR_PRIORITY)
 	wearer?.regenerate_icons()
-
```
Отсутствует модульная конструкция

Chituka replied:
Этого требует код. Прок, который я закомментил необходимо изменить, чтобы он вёл к иконкам ВНЕ кор-кода. Если убрать комментирование, то из-за дупликата появляются ошибки

RalseiDreemuurr commented on code/modules/clothing/outfits/factions/inteq.dm:
```diff
@@ -7,7 +7,11 @@
 	backpack = /obj/item/storage/backpack
 	satchel = /obj/item/storage/backpack/satchel
+	// [CELADON-REMOVE] - CELADON_OUTFIT
+	/*
```
Комментирование одной строки с помощью комментирования блоком? Зачем? Обычное используйте

RalseiDreemuurr commented:
Убрать вот это: 
![image](https://github.com/user-attachments/assets/359b10c6-25c0-4661-b473-306d7b715876)
Как? Меня не интересует. Главное, чтобы этого не было

Quinalion committed:
Отключил модпак с популяризацией МОДсьютов

RalseiDreemuurr commented:
![image](https://github.com/user-attachments/assets/33eb3d34-4951-4fc8-bf9b-4747750eabae)
![image](https://github.com/user-attachments/assets/e6b03614-7199-4c9f-b2fe-3e10ca73147d)
![image](https://github.com/user-attachments/assets/1f7e574a-f8d8-467e-88b7-8e5f9b20f415)
![image](https://github.com/user-attachments/assets/22d4c873-1960-45c9-ab92-355573fac9cd)
![image](https://github.com/user-attachments/assets/5232a2cf-2f67-457f-8164-0a4bcbba2657)
Код я написал и заметил сразу же косяк. На отключенном обычном интековском МОД-е спрайты сдвинуты на пиксель вправо, на включенной версии такого нет. Элитный не проверял

Chituka commented:
Твой код меняет интековскую версию на стандартную. Спрайты включенной версии не мои. Я пофиксил сдвиг пикселей, проблема была в спрайтах, но тебе тоже следует исправить код. А ещё это и есть элитный риг, обычный коричневого цвета, да..

RalseiDreemuurr commented:
После принятия этого ПР-а -> https://github.com/CeladonSS13/Shiptest/pull/1385
Нужно будет переместить спрайты в следующем порядке:
mod_celadon/_storge_icons/icons/mob/species/digitigrade_boots.dmi - для ботинок
mod_celadon/_storge_icons/icons/mob/species/digitigrade_suit.dmi - для костюмов (то бишь каркаса МОД-ов)
mod_celadon/_storge_icons/icons/mob/species/helms_snouted.dmi - для шлемов
В названиях спрайтов не должно быть никаких "-digi", "-snout" и прочего. Иначе будут использоваться обычные спрайты

Chituka committed:
Угу

Фикс доебов

Chituka committed:
Может это шизу гитхаба вылечит

Или это шиза дриммейкера, я хз

Chituka commented:
Будущим поколениям говорю. Удаление icon state из дми файла не убивает его полностью и от него что-то невидимое остается, что считается за изменение. Я хз как это

Chituka commented:
Пофиксил всё
Насчет ПРа и изменений при нем - это твоя работа, будем честны. Ты переносишь туда все спрайты, кроме, почему-то моих. Несмотря на то, что на альфу эти два ПРа, по факту, попадут в одно время

RalseiDreemuurr commented on code/modules/clothing/outfits/factions/inteq.dm:
```diff
@@ -7,7 +7,11 @@
+	[CELADON-REMOVE] - CELADON_OUTFIT
 	duffelbag = /obj/item/storage/backpack/duffelbag
+	[/CELADON-REMOVE] -
+	*/
```
Пишешь им делать одно, они делают совершенно другое
![image](https://github.com/user-attachments/assets/ded5575d-dbf2-468e-86ce-484d2f7cc600)
Плюсом теперь это не по правилам модульности

RalseiDreemuurr commented on code/modules/mod/mod_control.dm:
```diff
@@ -625,6 +625,7 @@
 		part.add_atom_colour(new_color, FIXED_COLOUR_PRIORITY)
 	wearer?.regenerate_icons()
 
+/* [CELADON-EDIT] - REASSIGNING ICON DERICTORY
```
Все ещё не по правилам модульности. Это не ЭДИТ, а РЕМУВ. 
![image](https://github.com/user-attachments/assets/a0d4a50c-1e3a-4b03-94b6-6fc6117f106b)

Chituka committed:
Merge branch 'CeladonSS13:beta-dev' into MOD_CARIB

Chituka committed:
*sigh

Chituka committed:
Merge branch 'MOD_CARIB' of https://github.com/Chituka/Shiptest into MOD_CARIB

Quinalion committed:
Теперь по правилам

Chituka committed:
Update inteq_caliban.dmm

Chituka committed:
Merge branch 'MOD_CARIB' of https://github.com/Chituka/Shiptest into MOD_CARIB

RalseiDreemuurr commented on code/modules/mod/mod_control.dm:
```diff
@@ -634,7 +634,10 @@
+		// [CELADON-EDIT] REASSIGNING ICONS DIRECTORY
+		//part.icon = used_skin[MOD_ICON_OVERRIDE] || 'icons/obj/clothing/modsuit/mod_clothing.dmi'
+		part.icon = used_skin[MOD_ICON_OVERRIDE] || 'mod_celadon/_storge_icons/icons/obj/modsuit/mod_clothing.dmi'
+		// [CELADON-EDIT] REASSIGNING ICONS DIRECTORY
```
Снова выполнено не по правилам модульности. Отсутствует название модпака и не подписан оригинальный код
![image](https://github.com/user-attachments/assets/2a84dd88-a4ed-483c-bc7c-4b3d27b41a25)

Chituka committed:
Part two

Xenophill my ass

Chituka committed:
Part one

Module code my ass

Chituka committed:
Merge https://github.com/CeladonSS13/Shiptest into MOD_CARIB

Chituka committed:
Caliban Updatery

Chituka commented:
+holopad to tadpole
-g36sh
+6.5 mm dmr
+firelocks between two engineering zone
+firelocks in some doors
+prettification
+additional ladder for caliban hangar

Chituka committed:
Updating Caliban and Tadpole

Chituka committed:
Update inteq_caliban.dmm

Chituka committed:
Merge branch 'CeladonSS13:beta-dev' into MOD_CARIB

Chituka committed:
Update inteq_tadpole.dmm

Chituka committed:
Update inteq_caliban.dmm
